src/commands/RunCommand.js

```diff
@@ -12,7 +13,20 @@ export default class RunCommand extends Command {
       return;
     }
 
-    this.packagesWithScript = this.packages
+    let packagesToRunCommandIn = this.packages;
+
+    if (this.flags.restrictTo && `${this.flags.restrictTo}`.length > 0) {
```

Comment:
I thought about that, but then `lerna run test --restrict-to "` would not run anything - if that is the expected behaviour I am happy to change it.